src/solver/simulation/timeseries-numbers.cpp

```diff
@@ -87,122 +87,56 @@ static bool GenerateDeratedMode(Study& study)
 };
 
 class loadAreaNumberOfTSretriever : public areaNumberOfTSretriever
 {
 public:
-    virtual ~loadAreaNumberOfTSretriever() = default;
-    loadAreaNumberOfTSretriever(Study& study) : areaNumberOfTSretriever(study)
-    {
-    }
-    std::vector<uint> getAreaTimeSeriesNumber(const Area& area)
-    {
-        std::vector<uint> to_return = {area.load.series.timeSeries.width};
-        return to_return;
-    }
-    uint getGeneratedTimeSeriesNumber()
+    std::vector<uint> getAreaTimeSeriesNumber(const Area& area) override
     {
-        return study_.parameters.nbTimeSeriesLoad;
+        return { area.load.series.timeSeries.width };
     }
 };
 
 class hydroAreaNumberOfTSretriever : public areaNumberOfTSretriever
 {
 public:
-    virtual ~hydroAreaNumberOfTSretriever() = default;
-    hydroAreaNumberOfTSretriever(Study& study) : areaNumberOfTSretriever(study)
-    {
-    }
-    std::vector<uint> getAreaTimeSeriesNumber(const Area& area)
-    {
-        std::vector<uint> to_return = {area.hydro.series->TScount()};
-        return to_return;
-    }
-    uint getGeneratedTimeSeriesNumber()
+    std::vector<uint> getAreaTimeSeriesNumber(const Area& area) override
     {
-        return study_.parameters.nbTimeSeriesHydro;
+        return { area.hydro.series->TScount() };
     }
 };
 
 class windAreaNumberOfTSretriever : public areaNumberOfTSretriever
 {
 public:
-    virtual ~windAreaNumberOfTSretriever() = default;
-    windAreaNumberOfTSretriever(Study& study) : areaNumberOfTSretriever(study)
-    {
-    }
-    std::vector<uint> getAreaTimeSeriesNumber(const Area& area)
-    {
-        std::vector<uint> to_return = {area.wind.series.timeSeries.width};
-        return to_return;
-    }
-    uint getGeneratedTimeSeriesNumber()
+    std::vector<uint> getAreaTimeSeriesNumber(const Area& area) override
     {
-        return study_.parameters.nbTimeSeriesWind;
+        return { area.wind.series.timeSeries.width };
     }
 };
 
 class solarAreaNumberOfTSretriever : public areaNumberOfTSretriever
 {
 public:
-    virtual ~solarAreaNumberOfTSretriever() = default;
-    solarAreaNumberOfTSretriever(Study& study) : areaNumberOfTSretriever(study)
-    {
-    }
-    std::vector<uint> getAreaTimeSeriesNumber(const Area& area)
-    {
-        std::vector<uint> to_return = {area.solar.series.timeSeries.width};
-        return to_return;
-    }
-    uint getGeneratedTimeSeriesNumber()
+    std::vector<uint> getAreaTimeSeriesNumber(const Area& area) override
     {
-        return study_.parameters.nbTimeSeriesSolar;
+        return { area.solar.series.timeSeries.width };
     }
 };
 
 class thermalAreaNumberOfTSretriever : public areaNumberOfTSretriever
 {
 public:
-    virtual ~thermalAreaNumberOfTSretriever() = default;
-    thermalAreaNumberOfTSretriever(Study& study) : areaNumberOfTSretriever(study)
-    {
-    }
-    std::vector<uint> getAreaTimeSeriesNumber(const Area& area)
+    std::vector<uint> getAreaTimeSeriesNumber(const Area& area) override
```

Comment:
Sure, but as you know, there are many changes to come from another PR, for example : **clusterCount()** is about to disappear from **thermal/container.h**, and other things.
To remove the code duplication you recommend, we could wait a bit, so that the part about clusters simplification is done.